eng/DotNetBuild.props

```diff
@@ -103,6 +104,14 @@
 
       <!-- Needed until https://github.com/dotnet/runtime/issues/109329 is fixed. -->
       <InnerBuildArgs Condition="'$(NetCoreAppToolCurrentVersion)' != ''">$(InnerBuildArgs) /p:NetCoreAppToolCurrentVersion=$(NetCoreAppToolCurrentVersion)</InnerBuildArgs>
+
+      <!--
+        If we're in a full-stack VMR build or Source-Build, publish all packages.
+        ShortStack jobs should only publish their specific packages.
+        In the future, some VMR jobs may not publish all packages.
+        SourceBuild will always need to publish everything.
+      -->
+      <InnerBuildArgs Condition="'$(ShortStack)' != 'true'">$(InnerBuildArgs) /p:PublishAllPackages=true</InnerBuildArgs>
```

Comment:
How is `PublishAllPackages` different from `EnableDefaultArtifacts`?

Reply:
I was trying to keep EnableDefaultArtifacts set per your feedback on a prior PR. We could change this to use that instead if you'd prefer.

Reply:
I don't think that I meant that it should always be used in every job, back in that other PR. The default experience should be to have it default to true (i.e. locally or on the designated primary vertical -> `win-64` which produces all the rid agnostic packages and passes the `-pack` switch in).

Could you please share a commit that shows how things would change when using `EnableDefaultArtifacts` as the control flow property?

Reply:
I've pushed an update to use `EnableDefaultArtifacts` here instead.